File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int[] ans = new int[2];

        int left = 0;
        int right = numbers.Length-1;

        while(left < right && right < numbers.Length){
            if((numbers[left] + numbers[right] )== target){
                ans[0] = left+1;
                ans[1] = right+1;

                return ans;
            }
            if((numbers[left] + numbers[right] )< target){
                left++;
            }      
            if((numbers[left] + numbers[right] ) > target){
                right--;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        oleft , oright = 0, len(matrix)-1
        while(oleft <= oright):
            omid = oleft + (oright-oleft) // 2
            if(matrix[omid][0] <= target <= matrix[omid][len(matrix[0])-1] ): 
                left , right= 0, len(matrix[0]) -1
                while(left <= right):
                    mid = left+(right-left)//2
                    if(matrix[omid][mid] == target): 
                        return True
                    elif(matrix[omid][mid] > target):
                        right = mid-1
                    else:
                        left = mid+1
                return False
                
            elif matrix[omid][0] > target:
                oright = omid - 1
            else:
                oleft = omid+1
        return False
        